### PR TITLE
Decorated timetable with popover and completed the form that adds an event

### PR DESCRIPTION
**Pull request by xuau34**
# Learned
- `setTimeout( function(){...}, 0)` is very useful that it will requeue the function at the end of execution queue.  
[reference](https://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful)
- jquery selector can apply for multiple filters.
E.g. `$(.row, .col)` - it contains `$(.row)` and `$(.col)`
- jquery selector can also chain the filters.
E.g. `$(.row.col)` - it will only contain `$(.col)` which is in `$(.row)`
- `$( document.createElement('div'))` is faster than other creating methods
- Listners can be binded to the same event multiple times, which may cause proplems, so the better way is to `.off` first, then to bind.
- For easier reading code, the plain text can be written as a string array, and then `.join('')`

# Overall what I've done

## Tag/Task in `Datastore`
Extended delete function so that we can pass the class directly

## Tag
Added new UUID property, which is unnecessary I think.
It will be deprecated later on.

## `TagServlet`
Automatically deleted the outdated tags (for eventDateTime) in `doGET`

## Timetable
螢幕快照 2019-07-19 下午3 49 30
* Added `hover` animation
* Added `bootstrap popover` to show more details
  Not fully completed yet.

螢幕快照 2019-07-19 下午3 52 39
* Added `dbclick` event to `timetable` that it will show a form to add events through `popover`
* Animated this `popover` -
  If users still `focus` on the form's input, it won't disappear but only when user focus on other things.
* Completed the form that it will only send requests with valid `datetime`

螢幕快照 2019-07-19 下午4 18 49
* Syncronized the form and `table-sample` so that it looks more active(?)

螢幕快照 2019-07-19 下午5 06 51
> tag update
>> the data that will be passed to `TagServlet`
>> the response

螢幕快照 2019-07-19 下午5 07 08
> Using `/dashboard/tags` to double check

Completed the links between adding events and tags

* **debug**
There's a wrong demand for `description` in `calendarServlet`'s `doPost`.
In other words, `calendarServlet` doesn't accept new event with `null description`, which is wrong.

# Plans
- Decorate the `table-sample` and `table-form`

### CODE REVIEW

**Review by ChungYujoyce — APPROVED** [review]
You are so incredible!!!!
